Fix MSVC2013 complains on Unicode characters

Chocobo1 opened the pull request:
Still one warning remains, but there isn't any unicode character at there:
```
gui\options_imp.cpp(621) : warning C4819: The file contains a character that cannot be represented in the current code page (950). Save the file in Unicode format to prevent data loss
```
I have no idea how to fix it...

sledgehammer999 commented:
@glassez any comment on this? Is this what you meant?

glassez commented:
Not quite it.
1. IMO, we should keep in a separate file only Unicode characters/words themselves. Ie C_INFINITY, C_UP, C_DOWN, C_LOCALE_JAPANESE, C_LOCALE_GEORGIAN - right. But C_COPYRIGHT, C_THANKS, etc. - is wrong. Instead, we need to keep here, for example
```c++
const char C_COPYRIGHT[] = "©";
```
and in the right (previous) place something like
```c++
trUtf8("Copyright %1 2006-2015 The qBittorrent project").arg(C_COPYRIGHT)
```
2. And yet. It is preferable to store string constants in the form of a **const char** array.

Chocobo1 commented:
>preferable to store string constants in the form of a const char array

Fixed.

>C_COPYRIGHT...

Fixed.

>C_THANKS, etc. - is wrong

I suppose you mean `C_THANKS` & `C_TRANSLATORS`, the reason I kept them this way is because there are many utf-8 characters in it and I'm not sure which one will cause compile error.
Even if I can find out every utf-8 char, the result could be messy, also, these strings are in HTML format, It's much easier to read/modify in the current way.
Maybe I get you wrong, then please enlighten me.

DoumanAsh commented:
Um... not all files are utf-8?

Chocobo1 commented:
Currently (before this PR) all files are in utf-8 without BOM.
In case you didn't follow: #3059.

DoumanAsh commented:
Aww... i didn't see it. So the problem is also with BOM...

sledgehammer999 commented:
After the pragma, leave a comment block giving enough info on the purpose/scope of the file.

glassez commented:
>I suppose you mean C_THANKS & C_TRANSLATORS, the reason I kept them this way is because there are many utf-8 characters in it and I'm not sure which one will cause compile error.
Even if I find out every utf-8 char, the result could be messy, also, these strings are in HTML format, it's much easier to read/modify in the current way.

I think we actually can store such large fragments in separate files (.html, for example) in application resources and load them using Qt, that work correctly with UTF8.

Chocobo1 commented:
Updated.

* There is already a `AUTHORS` file at the root directory and now I added `src/gui/thanks.html` & ` src/gui/translators.html`, looks like duplicate, any advice?

* I've added HTML headers to `src/gui/gpl.html` is that OK?

glassez commented:
>I added src/gui/thanks.html & src/gui/translators.html

I hope they use UTF8 without BOM.

Chocobo1 commented:
>I hope they use UTF8 without BOM.

Yes, they're UTF8 without BOM.

ngosang commented:
> @ngosang. In thanks.html & translators.html I simply added HTML headers, any suggestions?

HTML 5
```
<!DOCTYPE html>
<html>
<head>
    <meta charset="utf-8">
    <title>Translators</title>
</head>
...
```
HTML 4
```
<!DOCTYPE html PUBLIC "-//W3C//DTD HTML 4.01 Transitional//EN">
<html>
<head>
    <meta http-equiv="Content-Type" content="text/html; charset=UTF-8" />
    <title>Translators</title>
</head>
...
```
http://www.w3schools.com/tags/att_meta_charset.asp

In modern browsers both versions work without problem, just be consistent and use the same version in all files. One consideration, you are using **lang="en"**, this is ok and doesn't affect the codification but if the text is translatable I prefer to no specify the language like in the above examples.

Chocobo1 commented:
Updated, I removed the `lang` attribute.

sledgehammer999 commented:
Squash commits 97e1e81 and a69aa95 into one.

And unless someone objects I will merge.

Chocobo1 commented:
>Squash commits 97e1e81 and a69aa95 into one.

Done locally, will push when I finish all tasks.

Chocobo1 commented:
@sledgehammer999 updated.

sledgehammer999 commented:
Unfortunately I get corrupt text for the thanks and translators when I run the program. See pics.
![untitled](https://cloud.githubusercontent.com/assets/273315/8396384/3a38426e-1dab-11e5-9511-2f7f0ba0b527.png)
![untitled2](https://cloud.githubusercontent.com/assets/273315/8396386/3e7d7eac-1dab-11e5-87c2-a0953752e24c.png)

sledgehammer999 commented:
I left inline comments on how to fix the above.

Chocobo1 commented:
>use te_thanks->setHtml(QString::fromUtf8(thanksfile.readAll().constData())); 

I was thinking about the same thing. 
Code updated.

sledgehammer999 commented:
Thank you.

glassez commented:
We could load all of such string/symbols as thanks.html (from some file stored in resources; it can store, for example, a dictionary in JSON format as "CONST_NAME" => "CONST_VALUE"). It would have saved us from all sorts of perversions with the source files (such as BOM, undocumented #pragma, etc.).

sledgehammer999 commented:
>t would have saved us from all sorts of perversions with the source files (such as BOM, undocumented #pragma, etc.).

Thankfully this is limited to only 1 file. Your solution is cleaner in implementing but dirtier in using. (I am talking about the opening/parsing/extracting/assigning the strings from resource file).

glassez commented:
>I am talking about the opening/parsing/extracting/assigning the strings from resource file

This should be done only once when the program starts.
But I not insist. I hope that in some foreseeable future all compilers will support UTF8 and we will not need to rack their brains.

sledgehammer999 commented:
>I hope that in some foreseeable future all compilers will support UTF8 and we will not need to rack their brains.

Clearly only MSVC causes this pain. And it even needs a special pragma to work...
Let's see if msvc 2015 will remedy this.

glassez commented:
The problem is not whether this will be fixed in VS2015, but when we will cease to support older compilers. Now we still support VS2008.

sledgehammer999 commented:
>Now we still support VS2008.

In this case let me update what I said in this comment: https://github.com/qbittorrent/qBittorrent/issues/2433#issuecomment-74432239
I now use msvc2013 for the official releases since it also supports Windows XP. We can happily drop msvc2008.
And if msvc2015 will still support building for Windows XP we can happily drop msvc2013 too if needed,

glassez commented:
>I now use msvc2013 for the official releases since it also supports Windows XP. We can happily drop msvc2008.

Very good news. But we should post it, otherwise we will receive messages from those who will not be able to build qBittorrent using unsupported compilers.

P.S. For the sake of interest. How do you build Qt for WinXP using VS2013? I previously could not find such a possibility (to set compatibility with WinXP it was possible only from the IDE).

sledgehammer999 commented:
@glassez You're right, I need to update the wiki article about the extra steps. In the meantime:
For non gui things you only need to pass `/D "_USING_V110_SDK71_" /arch:IA32` to the compiler. Or for boost/libtorrent to the b2.exe : `define=_USING_V110_SDK71_ cxxflags=/arch:IA32`
For building gui you also need:
`/SUBSYSTEM:WINDOWS",5.01"` for the linker.

PS: I build qt myself so I do the changes inside the corresponding mkspec file, which in turn works for qbt too.

sledgehammer999 commented:
@Chocobo1 there is bug presenting itself with this. Take look at this: https://qbforums.shiki.hu/index.php/topic,3447.msg18350.html#msg18350

sledgehammer999 commented:
@Chocobo1 Nevermind. Fixed in https://github.com/qbittorrent/qBittorrent/commit/d59440f99ee10f1061777b9db7462fd4583a6e2a

glassez commented:
This PR broke my 4752fa5 commit changes. Please add ```<nobr></nobr>``` around **C++** in ```about.html```.

Chocobo1 commented:
>This PR broke my 4752fa5 commit changes. Please add `<nobr></nobr>` around C++ in `about.html`.

@sledgehammer999 , can you do this for us? Pushing this from your side is faster than I type this message. :p

sledgehammer999 commented:
>This PR broke my 4752fa5 commit changes. Please add <nobr></nobr> around C++ in about.html.

>@sledgehammer999 , can you do this for us? Pushing this from your side is faster than I type this message, thanks in advance. :p

Done: 626a3b10d3
